File: vistamovilpi/app/src/main/java/com/example/vista_movil_pi/modelo/Usuario.kt
package com.example.vista_movil_pi.modelo

data class Usuario(
    val nombre: String = "",
    val apellidos: String = "",
    val correo: String = "",
    val contra: String = "",
    val nif: String = "",
    val telefono: String = "",
    val direccion: String = "",
    val clientes: List<Cliente>,
) {
    // Constructor secundario que toma solo la lista de clientes

    val _id: String = ""
    constructor(clientes: List<Cliente>) : this("", "", "", "", "", "", "", clientes)
}

File: APIREST-MONGODB-CRM-PI/interfaces/usuario.ts
import { Document, Schema } from "mongoose";

export default interface IUsuario extends Document {
  nombre: Schema.Types.String;
  apellidos: Schema.Types.String;
  correo: Schema.Types.String;
  contra: Schema.Types.String;
  nif: Schema.Types.String;
  telefono: Schema.Types.String;
  direccion: Schema.Types.String;
  clientes: [Schema.Types.ObjectId];
}
